### PR TITLE
Update concurrency.md

### PR DESCRIPTION
**Pull request by dvdstelt**
Small updates about locking and rowversion, with link to our own documentation instead of Ayendes

### CODE REVIEW

File: nservicebus/sagas/concurrency.md [hunk 1]
```diff
@@ -32,19 +32,15 @@ NOTE: In NServiceBus Version 2 you had to create the constraint yourself in the
...
+### SQL Server
+When running using NHibernate persistance, the NServiceBus framework enables pessimistic locking, resulting in saga data being locked for a single thread. Adding ["Version" property to the saga data](http://docs.particular.net/nservicebus/nhibernate/saga-concurrency) gives NHibernate the option to work its magic and gain additional performance, especially in high volume messaging environments.
 
-Another option is to use a [transaction isolation level](https://msdn.microsoft.com/en-us/library/system.transactions.isolationlevel.aspx) of serializable but that causes [excessive locking](https://msdn.microsoft.com/en-us/library/ms173763.aspx) with considerable performance degradation.
+Another option is to use a [transaction isolation level](https://msdn.microsoft.com/en-us/library/system.transactions.isolationlevel.aspx) of serializable but that can cause [excessive locking](https://msdn.microsoft.com/en-us/library/ms173763.aspx) with considerable performance degradation, if not handled with care. Since NServiceBus 4 the default isolation level is "ReadCommitted".
```
Comment (dvdstelt) [inline nservicebus/sagas/concurrency.md]: Not sure if it's "ReadCommitted" for Sagas. There's a hint (updlock, rowlock) added, which perhaps should be described here as well?

File: nservicebus/sagas/concurrency.md [hunk 2]
```diff
@@ -32,19 +32,15 @@ NOTE: In NServiceBus Version 2 you had to create the constraint yourself in the
...
 
-NOTE: "Serializable" is the default isolation level for TransactionScopes.
-
-In NServiceBus Version 4 the default isolation level is
-"ReadCommitted", which is a more sensible default.
+NOTE: "Serializable" is the default isolation level for distributed transactions.
```
Comment (dvdstelt) [inline nservicebus/sagas/concurrency.md]: Not default for TransactionScope, but distributed transactions are promoted to serializable

Comment (ramonsmits) [inline nservicebus/sagas/concurrency.md]: @dvdstelt TransactionScope is something different than a DTC transaction. A TransactionScope can potentially be escalated to a fully distributed transaction when needed.

The default isolation level for a TransactionScope is also Serializable but we set that to ReadCommitted.

It is true that the default distributed transaction isolation level is serializable but that is not used due to it copying the selected isolation level from the scope. 

I don't see the behavior with a DTC transaction to be escalated to Serializable. It uses Read Committed as I get optimistic concurrency errors which could not be happening if the isolation level would be serializable as serializable should result in either sequential processing or deadlocking.

Comment (dvdstelt) [inline nservicebus/sagas/concurrency.md]: I know what a scope vs dtc is, but what I did not know is that default is serializable for scope. My bad. Will see if I should fix it after all. Thanks for the comment.

**Comment by SimonCropp** [comment]
@dvdstelt did u forget to delete the branch?

**Pull request closed without merging**